check_stability.py

```diff
@@ -353,6 +353,57 @@ def get_files_changed():
             for item in files[:-1].split("\0")]
 
 
+def get_affected_testfiles(files_changed):
+    affected_testfiles = []
+    # FIXME: treat ".xml" files as test files? or support files? or both? or neither?
+    testfile_extensions = [".html", ".htm", ".xhtml", ".xht", ".xml", ".svg"]
+    supportfile_extensions = [".js", ".json"]
+    for changedfile_pathname in files_changed:
+        # Skip any changed file that is actually a test file.
+        if os.path.splitext(changedfile_pathname)[1] in testfile_extensions:
+            continue
+        # Skip any changed file that is not a support file.
+        if os.path.splitext(changedfile_pathname)[1] not in supportfile_extensions:
+            continue
+        # We have a changed file that is a support file.
+        supportfile_name = os.path.basename(changedfile_pathname)
+        travis_root = os.path.join(os.path.abspath(os.curdir), "w3c", "web-platform-tests")
+        repo_path = changedfile_pathname[len(travis_root):]
+        os.path.normpath(repo_path)
+        path_components = repo_path.split(os.sep)[1:]
+        basedir = None
+        if len(path_components) == 2:
+            # Parent of this changed support file is a top-level subdir.
+            basedir = path_components[0]
+        elif len(path_components) > 2:
+            # This changed support file some number of subdirs down. So as
+            # the directory we walk to look for test files in, we use the
+            # parent directory of the directory this changed file is in.
+            basedir = os.path.dirname(os.path.dirname(repo_path))[1:]
```

Comment:
> I think a simpler algorithm would just be to look for the filename in any files in the same top-level directory.

Yes, changed so that we now start from the top-level directory and walk down looking for any test files that contain either the absolute path to the support file, or the relative path from that test file to that support file)